### ISSUE TITLE
Crash on reload in iOS simulator

### ISSUE THREAD

**Issue by kirillplatonov** [lazaronixon/react-native-turbolinks#19]
Thanks for the wonderful library. I have a small issue when using it.

I generated a new app using react-native CLI and added your library using instruction in the readme. After this, if I won't start turbolinks app in js and just will reload app in the simulator it will crash.

Here's my App.js file (just default RN code):
```
/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 * @flow
 */

import React, { Component } from 'react';
import {
  Platform,
  StyleSheet,
  Text,
  View
} from 'react-native';

const instructions = Platform.select({
  ios: 'Press Cmd+R to reload,\n' +
    'Cmd+D or shake for dev menu',
  android: 'Double tap R on your keyboard to reload,\n' +
    'Shake or press menu button for dev menu',
});

type Props = {};
export default class App extends Component<Props> {
  render() {
    return (
      <View style={styles.container}>
        <Text style={styles.welcome}>
          Welcome to React Native!
        </Text>
        <Text style={styles.instructions}>
          To get started, edit App.js
        </Text>
        <Text style={styles.instructions}>
          {instructions}
        </Text>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
  },
  welcome: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10,
  },
  instructions: {
    textAlign: 'center',
    color: '#333333',
    marginBottom: 5,
  },
});
```

This is how error looks like in Xcode:
<img width="1512" alt="screenshot" src="https://user-images.githubusercontent.com/839922/39703383-93cecb76-5210-11e8-951b-ed47f3a60130.png">

Log from Xcode:
```
2018-05-07 15:59:15.524 [info][tid:main][RCTCxxBridge.mm:210] Initializing <RCTCxxBridge: 0x6080001c4830> (parent: <RCTBridge: 0x6040000d3a90>, executor: (null))
2018-05-07 15:59:15.526068+0300 testapp[26340:1719060] Initializing <RCTCxxBridge: 0x6080001c4830> (parent: <RCTBridge: 0x6040000d3a90>, executor: (null))
2018-05-07 15:59:15.567 [warn][tid:main][RCTBridge.m:119] Class RCTCxxModule was not exported. Did you forget to use RCT_EXPORT_MODULE()?
2018-05-07 15:59:15.566754+0300 testapp[26340:1719060] Class RCTCxxModule was not exported. Did you forget to use RCT_EXPORT_MODULE()?
2018-05-07 15:59:15.578 [warn][tid:main][RCTModuleData.mm:67] Module RCTImageLoader requires main queue setup since it overrides `init` but doesn't implement `requiresMainQueueSetup`. In a future release React Native will default to initializing all native modules on a background thread unless explicitly opted-out of.
2018-05-07 15:59:15.578128+0300 testapp[26340:1719060] Module RCTImageLoader requires main queue setup since it overrides `init` but doesn't implement `requiresMainQueueSetup`. In a future release React Native will default to initializing all native modules on a background thread unless explicitly opted-out of.
2018-05-07 15:59:15.589 [info][tid:main][RCTRootView.m:293] Running application testapp ({
    initialProps =     {
    };
    rootTag = 1;
})
2018-05-07 15:59:15.667748+0300 testapp[26340:1719060] Running application testapp ({
    initialProps =     {
    };
    rootTag = 1;
})
2018-05-07 15:59:15.779335+0300 testapp[26340:1719864] [] nw_connection_get_connected_socket 4 Connection has no connected handler
2018-05-07 15:59:16.940 [info][tid:com.facebook.react.JavaScript] Running application "testapp" with appParams: {"rootTag":1,"initialProps":{}}. __DEV__ === true, development-level warning are ON, performance optimizations are OFF
2018-05-07 15:59:16.940292+0300 testapp[26340:1719854] Running application "testapp" with appParams: {"rootTag":1,"initialProps":{}}. __DEV__ === true, development-level warning are ON, performance optimizations are OFF
2018-05-07 15:59:16.953601+0300 testapp[26340:1719829] [] nw_connection_get_connected_socket 5 Connection has no connected handler
2018-05-07 15:59:17.787708+0300 testapp[26340:1719830] [] nw_connection_get_connected_socket 6 Connection has no connected handler
2018-05-07 15:59:18.541 [info][tid:main][RCTCxxBridge.mm:917] Invalidating <RCTCxxBridge: 0x6080001c4830> (parent: <RCTBridge: 0x6040000d3a90>, executor: (null))
2018-05-07 15:59:18.541290+0300 testapp[26340:1719060] Invalidating <RCTCxxBridge: 0x6080001c4830> (parent: <RCTBridge: 0x6040000d3a90>, executor: (null))
2018-05-07 15:59:18.541 [info][tid:main][RCTCxxBridge.mm:210] Initializing <RCTCxxBridge: 0x6080001c5460> (parent: <RCTBridge: 0x6040000d3a90>, executor: (null))
2018-05-07 15:59:18.541502+0300 testapp[26340:1719060] Initializing <RCTCxxBridge: 0x6080001c5460> (parent: <RCTBridge: 0x6040000d3a90>, executor: (null))
2018-05-07 15:59:18.542 [warn][tid:main][RCTModuleData.mm:67] Module RCTImageLoader requires main queue setup since it overrides `init` but doesn't implement `requiresMainQueueSetup`. In a future release React Native will default to initializing all native modules on a background thread unless explicitly opted-out of.
2018-05-07 15:59:18.542007+0300 testapp[26340:1719060] Module RCTImageLoader requires main queue setup since it overrides `init` but doesn't implement `requiresMainQueueSetup`. In a future release React Native will default to initializing all native modules on a background thread unless explicitly opted-out of.
(lldb)
```

The same problems appears when I reloading from any plain RN component. If turbolinks is initialized then it reloads everything fine.

Do you know what causing this problem? How I can fix it?

P.S. I use CMD + R for reloading in simulator.

**Comment by adanmayer** [comment]
I had the same problem, I've sent a pull request with a fix for this issue.

**Comment by lazaronixon** [comment]
@adanmayer Thank for you work... Could you provide a pull request with only fix for this problem?
Could you give me more explanation about startAppInView?

**Comment by adanmayer** [comment]
I've created a specific pull request for the bugfix. 
Sorry – I was working on a POC and added additional stuff, which is not related to the issue. 
I'm working on integrating the viewController into a different subView (other then the view of the root viewController). The basics are working, so it's iOS only for now. 
I will create a separate pull request once that is done. So you can decide, if you want to integrate that functionality ...

**Comment by lazaronixon** [comment]
Fixed with https://github.com/lazaronixon/react-native-turbolinks/releases/tag/v1.6.7

**Comment by lazaronixon** [comment]
It seens not working anymore, and I could not find why  😢

**Comment by lazaronixon** [comment]
#22 Fixed. It seems some compatibility problem with react-native-webview
